hbase-client/src/main/java/org/apache/hadoop/hbase/client/AsyncClientScanner.java

```diff
@@ -202,18 +208,24 @@ private long getPrimaryTimeoutNs() {
   private void openScanner() {
     incRegionCountMetrics(scanMetrics);
     openScannerTries.set(1);
+    final Context context = Context.current();
     addListener(timelineConsistentRead(conn.getLocator(), tableName, scan, scan.getStartRow(),
       getLocateType(scan), this::openScanner, rpcTimeoutNs, getPrimaryTimeoutNs(), retryTimer,
       conn.getConnectionMetrics()), (resp, error) -> {
-        if (error != null) {
-          consumer.onError(error);
-          return;
+        try (Scope ignored = context.makeCurrent()) {
+          if (error != null) {
+            consumer.onError(error);
+            return;
+          }
+          startScan(resp);
         }
-        startScan(resp);
       });
   }
 
   public void start() {
-    openScanner();
+    final Supplier<Span> spanSupplier = new TableOperationSpanBuilder(conn)
+      .setTableName(tableName)
+      .setOperation(scan);
+    TraceUtil.trace(() -> openScanner(), spanSupplier);
```

Comment:
So here, openScanner just returns void instead of a CompletableFuture, so after the openScanner call returns, we will end the span? I do not think this is the expected behavior? As in openScanner, we will store the current context and use the span in the context for all later callbacks?